gpii/node_modules/gsettingsBridge/gsettings_bridge.js

```diff
@@ -42,26 +42,31 @@
     });
 
     gpii.gsettings.getSingleKey = function (schemaId, key) {
-        return nodeGSettings.get_gsetting(schemaId, key);
+        var gSettings = new GSettings(schemaId);
+
+        return gSettings.get(key);
     };
 
     gpii.gsettings.setSingleKey = function (schemaId, key, value) {
-        nodeGSettings.set_gsetting(schemaId, key, value);
+        var gSettings = new GSettings(schemaId);
```

Comment:
Constructing a fresh instance for every settings operation seems messy and unnecessary. I think a clearer API would just fold these API methods into the module object itself with "schemaId" appearing in the first argument position, as it does here. In theory we lose out in consolidating the validation operations that you currently have written into the GSettings constructor, but I don't think there's a common usage pattern that would involve the module user working with a single schema for a set of related operations.

In other notes on the impl (here since there is no accompanying pull request for that), we should arrange to forward a setting of undefined to g_settings_reset rather than throw an exception - this would be consistent with usage in other settings handlers